File: 121. Best Time to Buy and Sell Stock/121. Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices):
        """
        :type prices: List[int]
        :rtype: int
        """
        if len(prices) < 2:
            return 0
        profit = 0
        minimum = prices[0]
        for i in prices:
            minimum = min(i, minimum)
            profit = max(i - minimum, profit)
        return profit

File: 121. Best Time to Buy and Sell Stock/121. Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        if (prices.empty()) {
            return 0;
        }
        int profit = 0;
        int buy = prices[0];
        for (int i = 0; i < prices.size(); i++) {
            //截至第i天，买入的最低价格：第i-1天的价格与第i天的价格的最小值
            buy = min(buy, prices[i]);
            //截至第i天，获得的最大利润：第i-1天的利润与在第i天卖出得到的最大利润
            profit = max(profit, prices[i] - buy);
        }
        return profit;
    }
};
